apps/web/src/utils/dateUtils.ts

```diff
@@ -5,12 +5,18 @@ import {
   formatDate,
   formDateStringToDate,
 } from "@gc-digital-talent/date-helpers";
+import { commonMessages } from "@gc-digital-talent/i18n";
 
 import { Maybe, Scalars } from "~/api/generated";
 
 export function formattedDate(date: Scalars["Date"], intl: IntlShape) {
   let dateString = date;
 
+  // handle accidental passing in nulls
+  if (date === null) {
+    return intl.formatMessage(commonMessages.notAvailable);
+  }
```

Comment:
null isn't type-wise allowed by the function as a value for `date`, but clearly that is leaking

Reply:
can we check for undefined also?

Reply:
when the error happened in prod , it had `undefined` value for the experience.awardedDate.
It survived that ?? check in the above line , went into formattedDate where it blew up.
I think we can manually set the value undefined and test this.

Reply:
Add testing? Sure, here https://github.com/GCTC-NTGC/gc-digital-talent/pull/8529/commits/7b38364ee9c38d9dfb898c11a1be8be8e892b855

Reply:
Awesome.. I was able to reproduce this error in my local. 
And verified this pr solved the issue. 
Before 
<img width="1675" alt="Screenshot 2023-11-20 at 10 36 50 AM" src="https://github.com/GCTC-NTGC/gc-digital-talent/assets/89482910/10788200-8507-4da9-b60a-2712cb7a1b6a">

After 
<img width="1462" alt="Screenshot 2023-11-20 at 10 41 56 AM" src="https://github.com/GCTC-NTGC/gc-digital-talent/assets/89482910/f86b5988-20b3-4e10-92d5-4ff9ac3ec500">

Reply:
Also career time line was also crashed without the date in award experience 
With this Pr , it can also load without date
<img width="833" alt="Screenshot 2023-11-20 at 10 45 19 AM" src="https://github.com/GCTC-NTGC/gc-digital-talent/assets/89482910/be67527d-715c-4e9f-8e00-5902a0cb94a0">